service/storage_service.cc

```diff
@@ -561,6 +566,66 @@ future<> storage_service::topology_change_coordinator_fiber(raft::server& raft,
 
         bool res;
         switch (node.rs->ring.value().state) {
+            case ring_slice::replication_state::commit_cdc_generation: {
+                // make sure all nodes know about new topology and have the new CDC generation data
+                // (we require all nodes to be alive for topo change for now)
+                std::tie(node, res) = co_await exec_global_command(std::move(node), raft_topology_cmd{raft_topology_cmd::command::barrier}, false, replaced_node);
+                if (!res) {
+                    break;
+                }
+
+                bool is_first_node = node.topology->normal_nodes.empty();
+                assert(!is_first_node || (raft.id() == node.id && node.topology->transition_nodes.size() == 1));
```

Comment:
If we're the first node in the cluster and we create the CDC generation timestamp, instead of choosing it in the future so it has time to propagate to other nodes before their clocks cross the timestamp, we can choose it to be the current time. This enables writes to CDC-enabled tables as soon as we finish booting (otherwise we'd have to wait until our clock crosses the timestamp, until that moment write attempts would throw an error that there's no CDC stream available for this write). But choosing the timestamp to be the current time is only safe if this is the first generation.

This logic was copy-pasted from the gossiper-bootstrap code, I guess a better way with the topology coordinator would be to check whether this is the first generation directly, i.e. if `topology->current_cdc_generation_id` is `nullopt`. I can change this in v2.